packages/docz-core/src/bundlers/webpack/config.ts

```diff
@@ -184,33 +189,50 @@ export const createConfig = (args: Args, env: Env) => async (
 
   const dev = !isProd
   const template = await htmlTemplate(args.indexHtml)
+  const createTemplate = (ctx: any) => {
+    const doc = parseHtml({ ctx, dev, template, config: args })
 
-  config.plugin('html-plugin').use(miniHtmlWebpack, [
-    {
-      context: {
-        ...args.htmlContext,
-        trimWhitespace: true,
-      },
-      template: (ctx: any) => {
-        const doc = parseHtml({ ctx, dev, template, config: args })
-
-        return dev
-          ? doc
-          : minify(doc, {
-              removeComments: true,
-              collapseWhitespace: true,
-              removeRedundantAttributes: true,
-              useShortDoctype: true,
-              removeEmptyAttributes: true,
-              removeStyleLinkTypeAttributes: true,
-              keepClosingSlash: true,
-              minifyJS: true,
-              minifyCSS: true,
-              minifyURLs: true,
-            })
+    return dev
+      ? doc
+      : minify(doc, {
+          removeComments: true,
+          collapseWhitespace: true,
+          removeRedundantAttributes: true,
+          useShortDoctype: true,
+          removeEmptyAttributes: true,
+          removeStyleLinkTypeAttributes: true,
+          keepClosingSlash: true,
+          minifyJS: true,
+          minifyCSS: true,
+          minifyURLs: true,
+        })
+  }
+
+  if (args.entries) {
+    Object.values(args.entries).forEach((entry, idx) => {
+      config.plugin(`html-plugin-${idx}`).use(miniHtmlWebpack, [
```

Comment:
`webpack-chain` doesn't seems to recognise multiple plugins with the same name or with multiple config objects in the array.